tools/webpack-plugin/src/BacktracePluginV4.ts

```diff
@@ -20,62 +22,113 @@ export class BacktracePluginV4 implements WebpackPluginInstance {
 
-                let debugId;
-                if (key.match(/.(c|m)?jsx?$/)) {
-                    debugId = crypto.randomUUID();
-                    assetDebugIds.set(key, debugId);
+                if (key.match(/\.(c|m)?jsx?$/)) {
+                    const debugId = crypto.randomUUID();
+                    const stats: AssetStats = { debugId };
+                    assetStats.set(key, stats);
+                    logger.log(`[${key}] generated debug ID ${debugId}`);
 
-                    source = this._sourceGenerator.addDebugIdToSource(source as never, debugId) as typeof source;
-                    source = this._sourceGenerator.addDebugIdCommentToSource(source as never, debugId) as typeof source;
+                    logger.time(`[${key}] inject source snippet`);
+                    try {
+                        source = this._sourceGenerator.addDebugIdToSource(source as never, debugId) as typeof source;
+                        logger.timeEnd(`[${key}] inject source snippet`);
+                        stats.sourceSnippet = true;
+                    } catch (err) {
+                        stats.sourceSnippet = err instanceof Error ? err : new Error('Unknown error.');
+                    }
+
+                    logger.time(`[${key}] inject sourcemap key`);
+                    try {
+                        source = this._sourceGenerator.addDebugIdCommentToSource(
+                            source as never,
+                            debugId,
+                        ) as typeof source;
+                        logger.timeEnd(`[${key}] inject sourcemap key`);
+                        stats.sourceComment = true;
+                    } catch (err) {
+                        stats.sourceComment = err instanceof Error ? err : new Error('Unknown error.');
+                    }
                 } else if (key.match(/\.(c|m)?jsx?\.map$/)) {
                     // The .map replacement should account for most of the use cases
                     const sourceKey = key.replace(/.map$/, '');
-                    debugId = assetDebugIds.get(sourceKey);
-                    if (!debugId) {
+                    const stats = assetStats.get(sourceKey);
+                    if (!stats) {
                         continue;
                     }
 
-                    source = this._sourceGenerator.addDebugIdToRawSourceMap(source as never, debugId) as never;
+                    logger.time(`[${key}] append sourcemap key`);
+                    try {
+                        source = this._sourceGenerator.addDebugIdToRawSourceMap(
+                            source as never,
+                            stats.debugId,
+                        ) as never;
+                        logger.timeEnd(`[${key}] append sourcemap key`);
+                        stats.sourceMapAppend = true;
+                    } catch (err) {
+                        stats.sourceMapAppend = err instanceof Error ? err : new Error('Unknown error.');
```

Comment:
the time command from line 66 will not end. Should we add a log error about that?